src/tyr/serializers.cc

```diff
@@ -209,6 +237,47 @@ json::ArrayPtr waypoints(const valhalla::Trip& trip) {
   return waypoints;
 }
 
+/*
+ * This function takes any waypoints (excluding origin and destination) and walks the shape
+ * for a match.  When a match is found, we store the geometry index.
+ * We use the valhalla midgard `equal` function to check for equality based on precision.
+ */
+json::ArrayPtr via_waypoints(valhalla::Options options, const std::vector<PointLL>& shape) {
+  // Create a vector of indexes based on the number of locations.
+  auto locs = *options.mutable_locations();
+  uint32_t i = 0;
+  std::vector<uint32_t> indexes;
+  for (const auto& loc : locs) {
+    indexes.push_back(i++);
+  }
+  auto via_waypoints = json::array({});
+  for (const auto& index : indexes) {
```

Comment:
Nit: Could `index` and `indexes` be more verbosively named, maybe something like `location_index`. There is a lot of indexing going on I feel like.

Reply:
Yes agree